iam-login-service/src/main/java/it/infn/mw/iam/core/IamTokenService.java

```diff
@@ -0,0 +1,47 @@
+package it.infn.mw.iam.core;
+
+import java.util.Set;
+
+import org.mitre.oauth2.model.OAuth2AccessTokenEntity;
+import org.mitre.oauth2.model.OAuth2RefreshTokenEntity;
+import org.mitre.oauth2.service.impl.DefaultOAuth2ProviderTokenService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Primary;
+import org.springframework.stereotype.Service;
+
+import com.google.common.collect.Sets;
+
+import it.infn.mw.iam.persistence.repository.IamOAuthAccessTokenRepository;
+import it.infn.mw.iam.persistence.repository.IamOAuthRefreshTokenRepository;
+
+@Service("defaultOAuth2ProviderTokenService")
+@Primary
+public class IamTokenService extends DefaultOAuth2ProviderTokenService {
+
+  private final IamOAuthAccessTokenRepository accessTokenRepo;
+  private final IamOAuthRefreshTokenRepository refreshTokenRepo;
+
+  @Autowired
+  public IamTokenService(IamOAuthAccessTokenRepository atRepo,
+      IamOAuthRefreshTokenRepository rtRepo) {
+    this.accessTokenRepo = atRepo;
+    this.refreshTokenRepo = rtRepo;
+  }
+
+
+  @Override
+  public Set<OAuth2AccessTokenEntity> getAllAccessTokensForUser(String id) {
+
+    Set<OAuth2AccessTokenEntity> results = Sets.newHashSet();
```

Comment:
In Mitre implementation, LinkedHashSet is used to preserve the resultset sorting.
Using HashSet, in the web interface tokens are listed in a different order than before.